Solutions/Menes.Hosting.AspNetCore/Menes/Internal/HttpRequestParameterBuilder.cs

```diff
@@ -172,6 +175,56 @@ public HttpRequestParameterBuilder(IEnumerable<IOpenApiConverter> converters, IL
             return parameters;
         }
 
+        private object? TryGetDefaultValueFromSchema(OpenApiSchema schema, IOpenApiAny? defaultValue)
+        {
+            return defaultValue switch
+            {
+                //// Convert to DateTimeOffset rather than use the DateTime supplied by default to be consistent with our converters.
+                OpenApiDate d       when schema.Format == "date"        => new DateTimeOffset(DateTime.SpecifyKind(d.Value, DateTimeKind.Utc)),
+                //// Convert to string rather than use the DateTimeOffset supplied by default to be consistent with our current (lack of) support for strings with format "date-time".
+                OpenApiDateTime dt  when schema.Format == "date-time"   => dt.Value.ToString("yyyy-MM-ddTHH:mm:ssK"),
+                OpenApiPassword p   when schema.Format == "password"    => p.Value,
+                OpenApiByte by      when schema.Format == "byte"        => by.Value,
+                OpenApiString s     when schema.Type == "string"        => this.ConvertValue(schema, s.Value),
+                OpenApiBoolean b    when schema.Type == "boolean"       => b.Value,
+                OpenApiLong l       when schema.Format == "int64"       => l.Value,
+                OpenApiInteger i    when schema.Type == "integer"       => i.Value,
+                OpenApiFloat f      when schema.Format == "float"       => f.Value,
+                OpenApiDouble db    when schema.Format == "double"      => db.Value,
+                OpenApiArray a      when schema.Type == "array"         => HandleArray(schema.Items, a),
+                OpenApiObject o     when schema.Type == "object"        => HandleObject(schema.Properties, o),
+                _ => throw new OpenApiSpecificationException("Default value for parameter not valid.")
+            };
+
+            object HandleArray(OpenApiSchema schema, OpenApiArray array)
```

Comment:
What's the thinking behind using local functions for `HandleArray` and `HandleObject`?

Reply:
We made it a function because the logic within would have been too messy to try and put within the switch expression. We made it a local function because nothing outside this method needs to use it.